packages/salesforcedx-vscode-core/test/vscode-integration/commands/forceCreateManifest.test.ts

```diff
@@ -0,0 +1,175 @@
+    });
+
+    afterEach(() => {
+      env.restore();
+    });
+
+    it('Should create first manifest from single sourceUri', async () => {
+      const packageXML = util.format(manifest, `<member>${class1}</member>`);
+      env.stub(ComponentSet, 'fromSource').returns({
+        getPackageXml: () => {
+          return packageXML;
+        }
+      });
+      env.stub(fs, 'existsSync').returns(false);
+      env.stub(fs, 'mkdirSync').returns(undefined);
+      env.stub(fs, 'writeFileSync').returns(undefined);
+      const packageName = 'package' + generateRandomSuffix() + '.xml';
+      inputBoxSpy.onCall(0).returns(packageName);
+      await forceCreateManifest(uri1, undefined);
+
+      expect(openDocSpy.calledOnce).to.equal(true);
+    });
+
+    it('Should create manifest from list of uris', async () => {
+      const packageXML = util.format(manifest, `<member>${class1}</member>\n<member>${class2}</member>\n`);
+      env.stub(ComponentSet, 'fromSource').returns({
+        getPackageXml: () => {
+          return packageXML;
+        }
+      });
+      env.stub(fs, 'existsSync')
+        .onFirstCall()
+        .returns(true)
+        .onSecondCall()
+        .returns(false);
+      env.stub(fs, 'writeFileSync').returns(undefined);
+      const packageName = 'package' + generateRandomSuffix() + '.xml';
+      inputBoxSpy.onCall(0).returns(packageName);
+      await forceCreateManifest(uri1, [uri1, uri2]);
+
+      expect(openDocSpy.calledOnce).to.equal(true);
+    });
+
+    it('Should create but not save manifest if cancelled', async () => {
+      const writeFileSpy = env.spy(fs, 'writeFileSync');
+      const packageXML = util.format(manifest, `<member>${class1}</member>\n<member>${class2}</member>\n`);
+      env.stub(ComponentSet, 'fromSource').returns({
+        getPackageXml: () => {
+          return packageXML;
+        }
+      });
+      env.stub(fs, 'existsSync')
+        .onFirstCall()
+        .returns(true)
+        .onSecondCall()
+        .returns(false);
+      inputBoxSpy.onCall(0).returns(undefined);
+      await forceCreateManifest(uri1, [uri1, uri2]);
+
+      expect(openDocSpy.calledOnce).to.equal(true);
+      expect(writeFileSpy.called).to.equal(false);
+    });
+
+  });
+
+  describe('Exception Handling', () => {
+
+    let env: sinon.SinonSandbox;
+    let openDocSpy: sinon.SinonSpy;
+    let inputBoxSpy: sinon.SinonStub;
+
+    before(() => {
+      env = createSandbox();
+    });
+
+    beforeEach(() => {
+      inputBoxSpy = env.stub(vscode.window, 'showInputBox');
+      openDocSpy = env.spy(vscode.workspace, 'openTextDocument');
+    });
+
+    afterEach(() => {
+      env.restore();
+    });
+
+    it('Should handle exception while creating component set', async () => {
+      env.stub(ComponentSet, 'fromSource').throws(
+        new Error(nls.localize('error_creating_packagexml'))
+      );
+      let exceptionThrown = false;
+      try {
+        await forceCreateManifest(uri1, [uri2]);
```

Comment:
can we use 
```
const manifestError = new Error(nls.localize('error_creating_packagexml'));
env.stub(ComponentSet, 'fromSource').throws(manifestError);
...
await expect(forceCreateManifest(uri1, [uri2]);).to.throw(manifestError);
```

it'd remove the need for the try/catch and extra checking of if an exception was thrown or not.

Reply:
@gbockus-sf It would be nice, but I think this failing because of the async function. After digging for a while, I found this which was similar to my issue: https://stackoverflow.com/questions/21587122/mocha-chai-expect-to-throw-not-catching-thrown-errors

Which got me to here: https://github.com/chaijs/chai/issues/882

So I think I'd have to write something like .rejected(), but I'm not quite sure how that would work. I'll push what I've got now, and we can discuss more, but I'd like to get this out sooner than get stuck too far in the syntax 😅 (although I would very much like to figure out how to get it to work!)

Reply:
@gbockus-sf I made this one with W-10522627, and feel free to add more details in the ticket with what you found.